plugins/com.google.cloud.tools.eclipse.appengine.facets/src/com/google/cloud/tools/eclipse/appengine/facets/FacetUninstallDelegate.java

```diff
@@ -95,4 +94,53 @@ protected IStatus run(IProgressMonitor monitor) {
     uninstallJob.schedule();
 
   }
+
+  private void removeAppEngineJarsFromMavenProject(IProject project, IProgressMonitor monitor) throws CoreException {
+    IMavenProjectFacade facade = MavenPlugin.getMavenProjectRegistry().getProject(project);
+    Model pom = facade.getMavenProject(monitor).getModel();
+
+    List<Dependency> currentDependencies = pom.getDependencies();
+    List<Dependency> updatedDependencies = updateMavenProjectDependencies(currentDependencies);
+    pom.setDependencies(updatedDependencies);
+
+    Properties properties = pom.getProperties();
+    updatePomProperties(properties, monitor);
+
+    DefaultModelWriter writer = new DefaultModelWriter();
+    File pomFile = pom.getPomFile();
+    try {
+      writer.write(pomFile, null /* options */, pom);
+    } catch (IOException e) {
+      throw new CoreException(StatusUtil.error(FacetUninstallDelegate.class, e.getMessage()));
```

Comment:
you can use `this` instead of `FacetUninstallDelegate.class` for `StatusUtil.error()` and then moving the code to other class wouldn't result in misleading error.